feat: upmerge from 2023-10-20

FabianGerke opened the pull request:
## Why:

Closes: #ID

## Describe your changes

## Checklist before requesting a review
- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

rbodnariu-plenty commented:
In order to run the PRs:
https://github.com/plentymarkets/module-webshop/pull/522
https://github.com/plentymarkets/module-mail-templates/pull/254/

must be merged into beta7 too.

ksted commented on packages/composables/src/useForgotPassword/index.ts:
```diff
@@ -1,21 +1,67 @@
-  UseForgotPasswordFactoryParams
+  sharedRef, useVSFContext
 } from '@vue-storefront/core';
+export const useForgotPassword = (id: string) => {
```
**issue**: Return type is missing.

ksted commented on packages/theme/pages/password-reset/_contactId/_hash.vue:
```diff
@@ -0,0 +1,151 @@
+
+    onMounted(async () => {
+      try {
+        email.value = await verifyHash(route.value.params.contactId, route.value.params.hash);
```
**issue**: Unnecessary call

We should be mindful of the number of REST calls we make. It would be preferable if we verified the hash when trying to actually reset a password and throw the error in that step.

We have to verify the hash again on the reset call itself anyway because it's possible that the has expires between opening the page and resetting the password.

renovate[bot] committed:
chore(deps): :arrow_up: Update devDependency @types/node to v18.18.4 (#159)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): :arrow_up: Update devDependency eslint to v8.51.0 (#160)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): :arrow_up: Update Yarn to v3.6.4 (#157)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump get-func-name from 2.0.0 to 2.0.2 (#156)

* chore(deps): bump get-func-name from 2.0.0 to 2.0.2

Bumps [get-func-name](https://github.com/chaijs/get-func-name) from 2.0.0 to 2.0.2.
- [Release notes](https://github.com/chaijs/get-func-name/releases)
- [Commits](https://github.com/chaijs/get-func-name/commits/v2.0.2)

---
updated-dependencies:
- dependency-name: get-func-name
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Szymon Dziewoński <szymon.dziewonski@gmail.com>

renovate[bot] committed:
chore(deps): :arrow_up: Update definitelyTyped (#161)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): :arrow_up: Update devDependency eslint-plugin-prettier to v5.0.1 (#162)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

FabianGerke committed:
Merge remote-tracking branch 'upstream/develop' into feat-upmerge-from-2023-10-20

FabianGerke committed:
fix: update yarn version

FabianGerke committed:
fix: update yarn.lock

FabianGerke committed:
Merge branch 'main' into feat-upmerge-from-2023-10-20
